### PR TITLE
Try GitHub Actions CI

### PR DESCRIPTION
**Pull request by hugovk**
Same as https://github.com/hugovk/pypistats/pull/68 but made from a non-fork branch.

cc @cclauss.

### CODE REVIEW

**Comment by hugovk** [comment]
Thanks again for the PRs, this is looking good. I've replicated what Azure Pipelines is doing, with the exception of uploading coverage to Codecov, and the sdist check. They can wait for later.

GitHub Actions looks very good. This is only using it as CI, and it's very like AP. Improvements:

* can include Python versions and OS in a matrix without creating templates and passing in parameters
* "[You can run up to 20 jobs concurrently per repository across all workflows](https://help.github.com/en/articles/about-github-actions#usage-limits)"! Compared to 10 with AP, 5 with Travis and 1 with AppVeyor.

---

https://help.github.com/en/articles/events-that-trigger-workflows#webhook-events says of `push`:

> Triggered on a push to a repository branch. Branch pushes and repository tag pushes also trigger webhook push events.

And `pull_request`:

> Triggered when a pull request is `assigned, unassigned, labeled, unlabeled, opened, edited, closed, reopened, synchronize, ready_for_review, locked, unlocked` or when a pull request review is requested or removed.

What a lot of events! 

https://help.github.com/en/articles/events-that-trigger-workflows#pull-request-events-for-forked-repositories says:

> When you open a pull request from a forked repository, the pull request opens on the base repository by default. In this scenario, GitHub sends the `pull_request` event to the base repository and no pull request events occur on the forked repository because the webhook configuration of the original base repository doesn't apply to forked repositories.

Does that mean `pull_request` is needed to trigger builds from forks? It continues:

> If you intend to use the `pull_request` event to trigger CI tests, we recommend that you set up your workflow configuration to listen to the `push` event.

Is that saying to use both `push` and `pull_request `? I guess so. Will send beta feedback. 

It also means for each event, there's 2 builds. 2 × (3 OS × 2 Python tests + 1 OS × 1 Python lint) = 2 × 7 = 14.

Still within the 20 limit here, but AP doesn't double up. 

---

Open questions

* How to trigger builds for PRs from forks?
* How to upload to Codecov? 
* How not to trigger builds for PR labels etc?
* How not to double up builds? 
* Does [CI skip] work?
* Cache pip downloads?
* How does job cancelling work when another job fails?

**Comment by cclauss** [comment]
@chrispat Any comments or recommendations on this exploration?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install --upgrade tox
```
Comment (cclauss) [inline .github/workflows/lint.yml]:
```
      python -m pip install --upgrade black mypy tox 
   - name: Format with black
      if: matrix.python-version >= 3.6
      run: |
        pip install black
        black --check . || true
    - name: Type check with mypy
      if: matrix.python-version >= 3.5
      run: |
        pip install mypy
        mypy --ignore-missing-imports .
```

Comment (hugovk) [reply .github/workflows/lint.yml]: Black is already run via tox and I'm not using mypy here. But thanks!

**Pull request merged**